### PR TITLE
ChartGraph: Tabular representation feature 

### PR DESCRIPTION
**Pull request by AlbertCarreras**
### Breking change

DefaultLabelProvider is expanded

#### What changed?

ChartGraph: Tabular representation feature 

Brave Browser - ChartGraph - Gestalt 2023-09-28 at 7 22 32 PM

#### Why?

[Accessibility. See https://github.com/pinterest/gestalt/pull/3219](https://deploy-preview-3219--gestalt.netlify.app/web/chartgraph#Tabular-representation)

### CODE REVIEW

File: packages/gestalt-charts/src/ChartGraph/TabularDataModalTable.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import { Table, Text, useDefaultLabel } from 'gestalt';
+import { type TransformedTabularDataType } from './useTabularData.js';
+
+type Props = {|
+  title: string,
+  isHorizontalLayout: boolean,
+  transformedTabularData: TransformedTabularDataType,
+  onSortChange: ('series' | 'x' | 'y') => void,
+  sortOrder: 'desc' | 'asc',
+  sortCol: null | 'series' | 'x' | 'y',
```
Comment (diyorbek) [inline packages/gestalt-charts/src/ChartGraph/TabularDataModalTable.js:12]: Can we extract these to union type declaration? They have been repeated a lot.

Comment (AlbertCarreras) [reply packages/gestalt-charts/src/ChartGraph/TabularDataModalTable.js]: DONE

File: packages/gestalt-charts/src/ChartGraph/useTabularData.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  isHorizontalLayout,
+}) => {
+  const { tableSeriesText, tableXAxisText, tableYAxisText } = useDefaultLabel('ChartGraph');
+
+  return `${tableSeriesText},${isHorizontalLayout ? tableXAxisText : tableYAxisText},${
+    isHorizontalLayout ? tableYAxisText : tableXAxisText
+  }\n${transformedTabularData
+    .map((x) => Object.values(x))
+    .map((e) => e.join(','))
+    .join('\n')}`;
```
Comment (diyorbek) [inline packages/gestalt-charts/src/ChartGraph/useTabularData.js:30]: nit: Do you think it's a good idea to wrap it with `useMemo`? I am not sure how often this hook is used.

File: packages/gestalt-charts/src/ChartGraph/useTabularData.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+const useTabularData: UseTabularDataProps = ({
+  data,
+  filterId,
+  filterOrder,
+  labelMap,
+  tickFormatter,
+  isHorizontalLayout,
+}) => {
+  const transformedTabularData = data
```
Comment (diyorbek) [inline packages/gestalt-charts/src/ChartGraph/useTabularData.js:92]: nit: Same concern here. These calculations are triggered in every render wherever the hook is present. Please, correct me if I'm not understanding something.

File: packages/gestalt-charts/src/ChartGraph/useTabularData.js (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt-charts/src/ChartGraph/useTabularData.js:30]: It´s a good concern. I can try to memoize. I think this is only called on the TabularDataModal component so there shouldn't be many rerenders in general after the mounting of Modal. If it were within ChartGraph, I think there could be more unrelated rerenders
@diyorbek

Comment (AlbertCarreras) [reply packages/gestalt-charts/src/ChartGraph/useTabularData.js]: DONE

File: packages/gestalt-charts/src/ChartGraph/useTabularData.js (hunk 3 above)
Comment (AlbertCarreras) [reply packages/gestalt-charts/src/ChartGraph/useTabularData.js]: DONE

**Review by diyorbek — APPROVED** [review]

**Pull request merged**